src/index.ts

```diff
@@ -29,4 +38,27 @@ const run = async () => {
 	}
 };
 
-void run();
+// This function's name is an exception from the naming convention
+// since we refer to the 1Password CLI here.
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const installCLI = async (): Promise<void> => {
+	await validateCli().catch(async () => {
```

Comment:
@edif2008 @jodyheavener Taking another look at this, I'm a bit confused by the expected behaviour here. We await the promise result of `validateCli` (imported from `op-js`), but we don't handle the result in a `.then()`; instead we `.catch()` the thrown error, and then we proceed to install the CLI?

Is this expected? Maybe I am misunderstanding how this is intended to work, but after a few times of re-reading this I'm unclear as to 1) why we expect `validateCli` to throw an error, 2) why we then do a bunch of stuff inside the error's `catch` block, and 3) why we don't do anything else post-`catch`.

Reply:
We use `validateCLI` to see if there's an installed 1Password CLI in the runner. If there is, we do nothing (i.e. there's no `then()` needed). If there is no CLI installed (which will throw an error), then we run the script to install it (which is done in the `catch()` part).

I will add an additional comment on this line to explain the flow for better clarity.

Reply:
Added with [2c980ca](https://github.com/1Password/load-secrets-action/pull/36/commits/2c980caffa1fb9062e8ea51dc82f36f0e0150393).

Reply:
I see, makes a lot more sense to me now. Thanks for the explanation + comment!